src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Logging/ProjectOutputWindowProjectLogger.cs

```diff
@@ -30,37 +29,19 @@ public ProjectOutputWindowProjectLogger(IProjectThreadingService threadingServic
             _outputWindowProvider = outputWindowProvider;
         }
 
-        public void WriteLine(string format, params object[] arguments)
-        {
-            if (_options.IsProjectOutputPaneEnabled)
-            {
-                // Only allocate if the pane is actually enabled
-                string text = string.Format(CultureInfo.CurrentCulture, format, arguments);
-
-                WriteLineCore(text);
-            }
-        }
-
         public void WriteLine(string text)
         {
-            if (_options.IsProjectOutputPaneEnabled)
-            {
-                WriteLineCore(text);
-            }
-        }
-
-        private void WriteLineCore(string text)
-        {
             // Extremely naive implementation of a Windows Pane logger - the assumption here is that text is rarely written,
             // so transitions to the UI thread are uncommon and are fire and forget. If we start writing to this a lot (such 
             // as via build), then we'll need to implement a better queueing mechanism.
             _threadingService.Fork(async () => {
+                if (await _options.GetIsProjectOutputPaneEnabledAsync())
```

Comment:
It was very deliberate that we did close to zero work when the logging is disabled. This and interpolated strings change has changed that - in the language service we log **a lot** - I want zero impact on that when logging is off.